### PR TITLE
Add documentation for spring-graphql integration.

### PR DESCRIPTION
**Pull request by srinivasankavitha**

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -19,7 +19,57 @@ You will also need JDK 17.
 If your application is on Spring Boot 2.7, you will need to use the 5.5.x release train of the DGS framework.
 If your application is on Spring Boot 2.6, you will need to use 5.4.x or earlier.
 
+
```
Comment (Emily) [inline docs/getting-started.md:22]: A couple thoughts on this section:
- `DGS Framework now using Spring GraphQL!` feels like an announcement, which I find tend not to age well in docs, unless they're very carefully scoped to things like release notes.
- Reading through I think it could be easy to get confused about which of these sections actually apply. Perhaps having a "Install Dependencies" header which can explain the difference, and then demoting spring-graphql and classic sections to a `###` header?

File: docs/getting-started.md [hunk 2]
```diff
@@ -130,8 +180,10 @@ This schema allows querying for a list of shows, optionally filtering by title.
 ## Implement a Data Fetcher
 
 Data fetchers are responsible for returning data for a query.
+With the new Spring-GraphQL integration, it is technically possible to mix and match the DGS/Spring-GraphQL programming models. 
+However, to maintain consistency in your codebase and to take full advantage of DGS features, we recommend sticking with the DGS programming model.
```
Comment (Emily) [inline docs/getting-started.md:184]: Where did we land on disabling the spring-graphql model by default? I'm still inclined to say it should be disabled by default, because it's hard for me to imagine why you would want to mix and match (with today's implementation) unless you were just migrating to plain spring-graphql. Otherwise, I suspect folks will just end up wondering why DGS features aren't working how they expect.

File: docs/spring-graphql-integration.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+## DGS Framework now using Spring GraphQL!
```
Comment (Emily) [inline docs/spring-graphql-integration.md:1]: Also feels very announcement-y. Perhaps we create a new announcements section? And add dates to the page?

**Review by Emily — CHANGES_REQUESTED** [review]

File: docs/getting-started.md (hunk 2 above)
Comment (srinivasankavitha) [reply docs/getting-started.md:184]: So here's the current thinking (which is different from what I had in mind a week ago in light of recent  experimentation). It's a bit wordy,  but wanted to write everything down so we can discuss the longer term strategy as well.

The integration in it's current state  does technically allows folks to define spring-graphql data fetchers/data loaders, but does not guarantee all DGS features will work in the context of spring-graphql. Therefore, we will strongly encourage folks to stick with defining DGS style data fetchers and data loaders when using the framework.
However, if folks choose to still experiment or use it for some reason, it is very much expected that not everything will work.  It is unlikely that users will accidentally mix DGS and spring graphql style controllers following our docs. If they do, it is probably intentional knowing that only spring-graphql features can be expected.

In the future, it seems reasonable to expect that we will directly integrate more with spring-graphql  as more features become available. We don't know what would be added in the future, but it would be good to have the option. As an example, after experimenting with the `GraphQlTester` (a similar offering to our `DgsQueryExecutor`),   it would be really nice to  allow users to take advantage of this feature ,since it does offer more options for testing. Already, this means we are opening up more of Spring GraphQl's functionality to be exposed more directly to users (not masked as DGS functionality).

In the long term, it also seems reasonable to work towards making all the DGS specific features  work for Spring GraphQl data fetchers and data loaders as well. This can be achieved either by asking Spring folks to offer similar hooks for e.g. data loader instrumentations, or we fill in the gaps for such features. Note that this is not required for the first release, but I can see that being future work for the next phase. 

So with that in mind, it may be beneficial to not disable spring-graphql data fetchers etc. but explicitly state that not everything will  work and that it is not supported.  Thoughts?

Comment (srinivasankavitha) [reply docs/getting-started.md:184]: Actually, I think it is worth pointing out that the main area that needs to be worked out is actually are the data loader concepts - metrics, additional instrumentation etc. The rest of the stuff might be less of a problem. So if we are moving towards that goal of being able to mix both styles of data fetchers/data loaders, this might be the main thing to focus on for the future.

File: docs/spring-graphql-integration.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+|----|----| ----- |
+| `dgs.graphql.schema-locations` | `spring.graphql.schema.locations` |  Use `dgs.graphql.schema-locations` |
+| N/A | `spring.graphql.schema.fileExtensions` | Not applicable, because `dgs.graphql.schema-locations` includes the path |
+| `dgs.graphql.graphiql.enabled` | `spring.graphql.graphiql.enabled` | Use `dgs.graphql.graphiql.enabled` |
+| `dgs.graphql.graphiql.path` | `spring.graphql.graphiql.path` | Use `dgs.graphql.graphiql.path` |
+| `dgs.graphql.websocket.connection-init-timeout` | `spring.graphql.websocket.connection-init-timeout` | DGS property sets the Spring-GraphQL property | 
+
+
+### File Uploads
+Support for file uploads will no longer be available by default in the DGS framework.
```
Comment (osi) [inline docs/spring-graphql-integration.md:82]: make sure this ends up in the release notes

File: docs/spring-graphql-integration.md
Comment (osi) [inline docs/spring-graphql-integration.md:1]: should observability-related changes be mentioned? will the traces and metrics the DGS framework captures be replaced by what Spring GraphQL does?

**Review by osi — APPROVED** [review]
overall looks good!

File: docs/spring-graphql-integration.md
Comment (srinivasankavitha) [reply docs/spring-graphql-integration.md:1]: So our internal Netflix metrics and traces remain unchanged. We only add micrometer based metrics for gql, which will still be published by the DGS framework, so no change there either.

Comment (osi) [reply docs/spring-graphql-integration.md:1]: ah, i was hoping that this change would replace the traces with what upstream provides, as it (very handily) incorporates the ability to test!

Comment (srinivasankavitha) [reply docs/spring-graphql-integration.md:1]: I'd be interested to hear more and can consider that as a future enhancement. We haven't explored that since we have been focused on making the existing features work as a first step.
